### PR TITLE
Adopt consistent approach for defining InternalsVisibleTo in the repo

### LINKED ISSUE #41244: Adopt consistent approach for defining InternalsVisibleTo in the repo
This is a follow-up to some PR feedback I received in https://github.com/dotnet/aspnetcore/pull/41238#discussion_r852307921 (and other PRs).

We use two different strategies for defining IVT in the repo at the moment:

- Property in the build
- Assembly attribute in source

For consistency, we might want to consider adopting one standard throughout the repo and using that everywhere.

### PR DESCRIPTION
**Pull request by silamon**
Fixes #41244. This pull request moves all InternalsVisibleTo from AssemblyInfo.cs to their respective project file (.csproj). When reviewing, compare them 2-by-2 since the csproj file is always next to the AssemblyInfo.cs file. 

One exception: the RateLimiting project had already the InternalsVisibleTo declared. In this project the AssemblyInfo.cs is removed without applying changes to .csproj project file.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=45748) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=45748) before we can accept your contribution.<br/><br/>:x: silamon [sign now](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=45748)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/aspnetcore?pullRequest=45748) it.</sub>

**Comment by Tratcher** [comment]
Wow, that's a bigger change than I expected. Good job.

Something's failing in DataProtection?
https://github.com/dotnet/aspnetcore/pull/45748/checks?check_run_id=10286080432

File: src/Servers/Kestrel/Transport.Sockets/src/Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.csproj [hunk 1]
```diff
@@ -32,4 +32,18 @@
     </EmbeddedResource>
   </ItemGroup>
 
+  <ItemGroup>
+    <InternalVisibleTo Include="Sockets.FunctionalTests" />
+    <InternalVisibleTo Include="Sockets.BindTests" />
+    <InternalVisibleTo Include="http2cat" />
+    <InternalVisibleTo Include="Microsoft.AspNetCore.Server.HttpSys.FunctionalTests" />
+    <InternalVisibleTo Include="http2cat" />
```
Comment (Tratcher) [inline src/Servers/Kestrel/Transport.Sockets/src/Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.csproj:40]: Duplicate
```suggestion
```

**Comment by silamon** [comment]
> Something's failing in DataProtection? https://github.com/dotnet/aspnetcore/pull/45748/checks?check_run_id=10286080432

Yes, thanks for looking. Actually, it's not only in DataProtection. There are also a few cases in Kestrel. 
Strange thing is that I've included the correct InternalsVisibleTo in the project. At least I think so, but I may miss some knowledge. :)

For example: 
2022-12-24T11:26:45.6372501Z D:\a\_work\1\s\src\Servers\Kestrel\Kestrel\src\WebHostBuilderKestrelExtensions.cs(37,76): error CS0122: 'KestrelServerOptionsSetup' is inaccessible due to its protection level [D:\a\_work\1\s\src\Servers\Kestrel\Kestrel\src\Microsoft.AspNetCore.Server.Kestrel.csproj]

There's a InternalsVisibleTo declared in Microsoft.AspNetCore.Server.Kestrel.Core for Microsoft.AspNetCore.Server.Kestrel. So, I'm not really sure why it remains inaccessible. The namespace of KestrelServerOptionsSetup does have "Internal" and most of those DataProtection classes also have that, but this is more of an assembly thing if I understand correctly. 

May it have something to do with the fact that I omitted the PublicKey?

**Comment by silamon** [comment]
Nevermind, I think I saw the problem.

**Review by Tratcher — APPROVED** [review]

**Comment by JamesNK** [comment]
A lot of Moq and Castle.DynamicProxy tests are failing.

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**